src/mongocxx/bulk_write.cpp

```diff
@@ -33,8 +33,16 @@ bulk_write& bulk_write::operator=(bulk_write&&) noexcept = default;
 
 bulk_write::~bulk_write() = default;
 
-bulk_write::bulk_write(bool ordered)
-    : _impl(stdx::make_unique<impl>(libmongoc::bulk_operation_new(ordered))) {
+bulk_write::bulk_write(options::bulk_write options)
+    : _impl(stdx::make_unique<impl>(libmongoc::bulk_operation_new(options.ordered()))) {
+    auto options_wc = options.write_concern();
```

Comment:
I think what we have now is pretty good. The `::bulk_write` constructor will default construct an `options::bulk_write` object, and that object defaults to ordered (which is the only safe default). I think you should know at the time you are constructing the holder for the individual ops whether you are planning those to have ordered or unordered semantics.